### ISSUE TITLE
Clarification on the linear probing for semantic segmentation

### ISSUE THREAD

**Issue by Antuke** [facebookresearch/perception_models#69]
Hello, thank you for your work and for making it open-source. I would like to have a clarification on one of the paragraph of the paper.

In the paper it's said that semantic segmentation is achieved using the following approach:

> Semantic Segmentation. For semantic segmentation, we evaluate our pretrained models on ADE20K [167]
> semantic segmentation task. We use a linear layer and convolutional layer to map intermediate spatial features
> to segmentation masks following [98]. The models are evaluated and then features are resized to 518 × 518.
> We only use features from single layer. The probing layers are finetuned with AdamW [83] with a learning
> rate of 0.001. 

What is not clear to me is how is the convolutional layer is used? It seems to me that in the dinov2 paper [98] cited there is no specific mention of use of a convolution layer, has its said:

> "a linear layer is trained to predict class logits from a patch tokens. It is used to produce a lowresolution logit map (eg 32x32 for a model with patch size 16), which is then upsampled to full resolution
> (512x512) to obtain a segmentation map."

My guess is that it's some kinda of interpolation + convolution with stride set 1, but I would like to hear from you to clarify this doubt.

**Comment by dbolya** [comment]
Thanks for bringing this up. I just checked the actual runs themselves and they _didn't_ use an additional layer. The decoder head is a _batch norm_ + a 1x1 conv, not a linear and a conv. Sorry for the confusion, I'll fix that in the current draft so that it's fixed next time we update the paper.

So, same as DINO (the conv being a linear layer in disguise). I believe DINO also uses a batch norm in the original code if I'm not mistaken? We actually implemented this in mmseg not the DINOv2 repo, so there may be some subtle differences.

**Comment by Antuke** [comment]
Thank you for your response 😊
